specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,175 @@
+3. It must be able to return the measured value for each qubit in the order requested by the program. This can be done in software as a post processing step after all quantum instructions and the final measurements have been performed; it does not require hardware or runtime support.
+
+TODO: edit the third bullet to just say that it needs to be able to produce one of the define output formats? Are all output formats equivalent (I think so, *if* the labeling scheme is defined)?  
+-> "the order requested by the program" may be somewhat ambiguous in any case, if we want to allow for parallelism/async execution
+
+The second requirement should be taken to mean that a Base Profile compilant program does *not* apply instructions to a qubit after it has been measured; all instructions to transform the quantum state can be applied before performing any measurements. It specifically also means that there is no need for the QPU to be able to measure only a subset of all available qubits at a time.
+
+## Program Structure
+
+A Base Profile compliant program is defined in the form of a single LLVM bitcode file that consists of the following:
+- definition of the opaque `Qubit` and `Result` types
+- [entry point definition](#entry-point-definition)
+- declarations of [QIS functions](#quantum-instruction-set)
+- declarations of functions used for [output recording](#output-recording)
+- [attributes](#attributes)
+
+The human readable LLVM IR for the bitcode can be obtained using standard 
+[LLVM tools](https://llvm.org/docs/CommandGuide/llvm-dis.html). For the purpose of clarity, this specification contains examples of the human readable IR emitted by [LLVM 13](https://releases.llvm.org/13.0.1/docs/LangRef.html). While the bitcode reprentation is portable and usually backward compatible, there may be visual differences in the human readable format depending on the LLVM version. This differences are irrelevant when using standard tools to load, manipulate, and/or execute bitcode.
+
+The code below illustrates how a simple program looks like within a Base Profile representation:
+
+```llvm
+; type definitions
+
+%Result = type opaque
+%Qubit = type opaque
+
+; global constants (labels for output recording)
+
+@0 = internal constant [16 x i8] c"label_schema_id\00"
+@1 = internal constant [3 x i8] c"r1\00"
+@2 = internal constant [3 x i8] c"r2\00"
+
+; entry point definition
+
+define i64 @Entry_Point_Name() #0 {
+entry:
+
+  ; calls to QIS functions  
+  tail call void @__quantum__qis__h__body(%Qubit* null)
+  tail call void @__quantum__qis__cnot__body(%Qubit* null, %Qubit* inttoptr (i64 1 to %Qubit*))
+  tail call void @__quantum__qis__mz__body(%Qubit* null, %Result* null)
+  tail call void @__quantum__qis__mz__body(%Qubit* inttoptr (i64 1 to %Qubit*), %Result* inttoptr (i64 1 to %Result*))
+
+  ; calls to record the program output
+  tail call void @__quantum__rt__initialize_record_output(i8* getelementptr inbounds ([16 x i8], [16 x i8]* @0, i32 0, i32 0))  
```

Comment:
Under the assumption that the base profile should represent the minimal instruction set any one of the current providers have, `getelementptr` is not supported.

Likewise, `__quantum__rt__initialize_record_output` is not currently [listed](https://github.com/qir-alliance/qat/blob/main/qir/qat/ValidationPass/ValidationPassConfiguration.cpp) in any profile.

Reply:
Yes, these are deviations from/updates to the current implementation that we (this workstream) suggest.